File: frontend/time-planner/src/app/time-tracking/models/work-item.ts
export interface WorkItem{
  id: string,
  name:string,
  durations: Duration[],
  category: string,
  nextTime: string | undefined,
  cronExpression: string | undefined,
  recurrenceStartsOn: string | undefined,
  recurrenceEndsOn: string | undefined,
  isAfterPreviousCompleted: boolean | undefined,
  maxRepetetionsCount: number | undefined,
  isOnPause: boolean | undefined,
  sortOrder: number,
  completedAt: string | undefined
}

export interface Duration{
  date: string,
  value: string
}

export interface WorkItemUpdateRequest{
  id: string,
  name: string,
  category: string,
  updateRecurrence: boolean,
  cronExpression: string | undefined,
  recurrenceStartsOn: string | undefined,
  recurrenceEndsOn: string | undefined,
  isAfterPreviousCompleted: boolean | undefined,
  maxRepetetionsCount: number | undefined,
  isOnPause: boolean | undefined,
  sortOrder: number
}

File: backend/TimePlanner.Domain/Models/WorkItem.cs
namespace TimePlanner.Domain.Models
{
  /// <summary>
  /// Work item.
  /// </summary>
  public record struct WorkItem(
    Guid? Id,
    string Name,
    Category Category,
    DateTime CreatedAt,
    DateTime? CompletedAt,
    DateTime? NextTime,
    string? CronExpression,
    DateTime? RecurrenceStartsOn,
    DateTime? RecurrenceEndsOn,
    bool? IsIfPreviousCompleted,
    int? MaxRepetitionCount,
    int? RepetitionCount,
    bool? IsOnPause,
    int SortOrder,
    List<Duration> Durations);

  public record struct Duration(int? id, DateOnly Date, TimeSpanValue Value);

  public enum Category
  {
    Today,
    Tomorrow,
    NextWeek,
    Completed,
    Scheduled,
    Archived
  }
}
